waspc/src/Wasp/Project/WaspFile.hs

```diff
@@ -117,9 +118,23 @@ compileWaspTsFile waspProjectDir tsconfigNodeFileInWaspProjectDir waspFilePath =
       ( runNodeCommandAsJob
           waspProjectDir
           "npx"
+          -- We're using tsc to compile the *.wasp.ts file into a JS file.
+          --
+          -- The compilation rules mostly come from tsconfig.wasp.json but also
+          -- include several overrides:
+          --   - We don't keep all the rules in tsconfig.wasp.json because it
+          --   would give users a way to break things.
+          --   - We don't keep all the rules here (inline) because users would
+          --   suffer bad DX (no IDE support in *.wasp.ts file).
```

Comment:
I found this confusing to read, had to read it a couple of times to understand what it is about.

The problem was that you said "include several overrides:" and then bullet points, but those bullet points were not overrides, as I expected, but instead they were reasons why it makes sense to have these overrides. Which I don't think is what a reader expects, so I was reading and reading trying to undersatnd how is this connected together.

I would instead say that yes, the main source of rules is tsconfig.wasp.json, but we add some additional rules on top of that. And reason for that is that we don't want  to keep all these rules in ... because of ..., and also is not all here because ... .

Although I think better way to put it would be to just say that rules in tsconfig.wasp.json are aimed at user and providing them with the best IDE support, while here we have a bit different requirements, and that is the reason for additional rules we add on top.